### PR TITLE
Use Admiral CI workflows

### PR DESCRIPTION
**Pull request by cicdguy**
Use common Admiral CI workflows.
Relates to #1374

### CODE REVIEW

**Comment by cicdguy** [comment]
An admin will need to change these settings in the branch protection settings, since the workflow names will change as a result of these common workflows.

Screen Shot 2022-08-26 at 11 21 46 AM

File: .github/workflows/common.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  docs:
+    name: Documentation
+    uses: pharmaverse/admiralci/.github/workflows/pkgdown.yml@main
+    if: github.event_name == 'push'
+    with:
+      r-version: $R_VERSION
+      # Whether to skip multiversion docs
+      # Note that if you have multiple versions of docs,
+      # your URL links are likely to break due to path changes
+      skip-multiversion-docs: true
```
Comment (cicdguy) [inline .github/workflows/common.yml:80]: We will set this to `false` following the 0.9.0 release.

File: .github/workflows/common.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  # to run this workflow on demand, anytime
+  workflow_dispatch:
+  # 'push' events are triggered when commits
+  # are pushed to one of these branches
+  push:
+    branches:
+      - main
+      - devel
+      - pre-release
+      - test
```
Comment (thomas-neitmann) [inline .github/workflows/common.yml:18]: We don't have a `test` branch.

File: .github/workflows/common.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  # 'pull_request' events are triggered when PRs are
+  # created against one of these target branches.
+  pull_request:
+    branches:
+      - main
+      - devel
+      - pre-release
+      - test
```
Comment (thomas-neitmann) [inline .github/workflows/common.yml:26]: See comment above.

File: .github/workflows/common.yml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      # Setting to 'false' will require you to create
+      # an orphan branch called 'badges' in your repository
+      skip-coverage-badges: false
+  templates:
+    name: Check Templates
+    uses: pharmaverse/admiralci/.github/workflows/check-templates.yml@main
+    if: >
+      github.event_name == 'push' || github.event_name == 'pull_request'
+    with:
+      r-version: $R_VERSION
```
Comment (thomas-neitmann) [inline .github/workflows/common.yml:109]: Right now we only run this when a PR is approved as this takes up quite a bit of run time (and thus our free GA minutes).

File: .github/workflows/common.yml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  validation:
+    name: Validation
+    uses: pharmaverse/admiralci/.github/workflows/r-pkg-validation.yml@main
+    if: github.event_name == 'release'
+    with:
+      r-version: $R_VERSION
+  check:
+    name: Check
+    uses: pharmaverse/admiralci/.github/workflows/r-cmd-check.yml@main
+    if: github.event_name == 'pull_request'
```
Comment (thomas-neitmann) [inline .github/workflows/common.yml:70]: I assume that still runs on multiple R versions? What about the package versions? In our current `R CMD check` workflow we use MRAN snapshots from the date of the release of the tested R version. I'd very much like to continue doing that going forward.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]
Thanks @cicdguy! Please take a look at my comments. In addition please merge in the latest changes from `devel`. That should fix the failing `Check Templates` workflow.

**Review by thomas-neitmann — APPROVED** [review]

**Pull request merged**